app/controllers/api/users_controller.rb

```diff
@@ -12,24 +12,15 @@ def index
 
     @target = target_allowlist.include?(params[:target]) ? params[:target] : 'student_and_trainee'
 
-    target_users =
-      if @target == 'followings'
-        current_user.followees_list(watch: @watch)
-      elsif @tag
-        User.tagged_with(@tag)
-      elsif @company
-        User.where(company_id: @company).users_role(@target)
-      elsif @target.in? %w[hibernated retired]
-        User.users_role(@target)
-      else
-        User.users_role(@target).unhibernated.unretired
-      end
+    users = target_users
+    users.order(:last_activity_at) if @target == 'inactive'
+    @users = users
+             .preload(:company, :avatar_attachment, :course, :tags)
+             .order(updated_at: :desc)
+             .page(params[:page])
+             .per(PAGER_NUMBER)
 
-    @users = target_users.page(params[:page]).per(PAGER_NUMBER)
-                         .preload(:company, :avatar_attachment, :course, :tags)
-                         .order(updated_at: :desc)
-
-    @users = search_for_users(@target, target_users, params[:search_word]) if params[:search_word]
+    @users = search_for_users(@target, users, params[:search_word]) if params[:search_word]
```

Comment:
```suggestion
  @users = 
    if params[:search_word]
      search_for_users(@target, users, params[:search_word]) 
    else
        users
             .preload(:company, :avatar_attachment, :course, :tags)
             .order(updated_at: :desc)
             .page(params[:page])
             .per(PAGER_NUMBER)
    end
```

みたいにした方がよいのでは（インデントは修正してください）